partiql-eval/src/main/kotlin/org/partiql/eval/internal/Environment.kt

```diff
@@ -30,9 +30,10 @@ internal class Environment {
     }
 
     /**
-     * Gets the scope/record/variables-environment at the requested [index].
+     * Gets the scope/record/variables-environment at the requested [depth].
      */
-    operator fun get(index: Int): Record {
+    operator fun get(depth: Int): Record {
+        val index = scopes.lastIndex - depth
```

Comment:
Scope size is known at compile time so the index can be precomputed in Compiler.kt.


So ExprVarOuter looks like before

```
class ExprVarOuter(
    val scope: Int,
    val ref: Int,
    val env: Env,
)
      // env[scope][ref]
```

And you compute `scope` in Compiler.kt